Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -0,0 +1,300 @@
+        {
+          isEnabled = false;
+        }
+      }
+      // TODO: nullcheck
+      var item = document.getElementById (itemInfo.ID) as HTMLElement;
+      let anchor = $ (item).children().eq (0)[0] as HTMLAnchorElement;
+      var icon = $ (anchor).children().eq (0)[0] as HTMLImageElement;
+      if (isEnabled)
+      {
+        if (icon != null && icon.nodeType == 1)
+          icon.src = itemInfo.Icon;
+        item.className = this._itemClassName;
+        if (itemInfo.Href != null)
+        {
+          if (itemInfo.Href.toLowerCase().indexOf ('javascript:') >= 0)
+          {
+            anchor.href = '#';
+            anchor.removeAttribute ('target');
+            anchor.setAttribute ('javascript', itemInfo.Href);
+            anchor.removeAttribute ('onclick');
+            // TODO: nullcheck anchor.getAttribute ('javascript')
+            anchor.onclick = function () { eval (anchor.getAttribute ('javascript')!); return false; };
+          }
+          else
+          {
+            anchor.href = itemInfo.Href;
+            if (itemInfo.Target != null)
+              anchor.target = itemInfo.Target;
+            anchor.removeAttribute ('javascript');
+            anchor.removeAttribute ('onclick');
+            anchor.onclick = null;
+          }
+        }
+        anchor.removeAttribute ('aria-disabled');
+      }
+      else
+      {
+        if (icon != null && icon.nodeType == 1)
+        {
+          if (itemInfo.IconDisabled != null)
+            icon.src = itemInfo.IconDisabled;
+          else
+            icon.src = itemInfo.Icon;
+        }
+        item.className = this._itemDisabledClassName;
+        anchor.removeAttribute ('href');
+        anchor.removeAttribute ('target');
+        anchor.removeAttribute ('javascript');
+        anchor.removeAttribute ('onclick');
+        anchor.onclick = null;
+        anchor.setAttribute ('aria-disabled', 'true');
+      }
+
+      if (itemInfo.DiagnosticMetadata)
+      {
+        // Do not render empty diagnostic metadata attributes
+        $.each(itemInfo.DiagnosticMetadata, function (key, value)
+        {
+          if (value === "" || value === null)
+          {
+            delete itemInfo.DiagnosticMetadata[key];
+          }
+        });
+
+        $(item).attr(itemInfo.DiagnosticMetadata);
+      }
+
+      if (itemInfo.DiagnosticMetadataForCommand)
+      {
+        // Do not render empty diagnostic metadata attributes
+        $.each(itemInfo.DiagnosticMetadataForCommand,
+          function (key, value) {
+            if (value === "" || value === null)
+            {
+              delete itemInfo.DiagnosticMetadataForCommand[key];
+            }
+          });
+
+        itemInfo.DiagnosticMetadataForCommand['data-is-disabled'] = isEnabled ? 'false' : 'true';
+
+        $(anchor).attr(itemInfo.DiagnosticMetadataForCommand);
+      }
+    }
+  }
+
+  public static OnKeyDown (event: JQueryKeyEventObject, $listMenu: JQuery): void
```

Comment:
Is this only needed from within the own class or also from other places?

Reply:
I only found references to this function within this file, no idea about downstream projects though, that's why I left the global function as-is.

Reply:
we may assume they're unsed.